### ISSUE TITLE
How do I run inference? Can't find the model

### ISSUE THREAD

**Issue by kdkalp** [shijing001/text_classifiers#3]
Hello,
Congratulations on putting together such a nice tool. I read the paper, and while I am just a beginner in this field, I am quite excited to try LESA - BERT on some of the patient messages in developing countries. 

Sorry for being a noob - can you point me to the pre-trained weights for LESA-BERT? I couldn't find them in the repository. I saw that you couldn't share the data because of privacy issues, so  I am hoping that you could share the model trained on your data.

**Comment by shijing001** [comment]
The initial weights are the BERT-base weights pre-trained on large uncased English corpora.
https://huggingface.co/bert-base-uncased

Best
shijing

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

From: ***@***.***>
Sent: Tuesday, May 11, 2021 4:52 PM
To: ***@***.***>
Cc: ***@***.***>
Subject: [shijing001/text_classifiers] How do I run inference? Can't find the model (#3)


Hello,
Congratulations on putting together such a nice tool. I read the paper, and while I am just a beginner in this field, I am quite excited to try LESA - BERT on some of the patient messages in developing countries.

Sorry for being a noob - can you point me to the pre-trained weights for LESA-BERT? I couldn't find them in the repository. I saw that you couldn't share the data because of privacy issues, so I am hoping that you could share the model trained on your data.

**Comment by datakalp** [comment]
Dear Shijing,

Thank you for the prompt response. 
I see the pointer to the BERT - base model. What I am looking for is the LESA BERT model (Distil-LESA-BERT-6) for urgency-based classification that is referred in your paper. I can get some data of text samples with urgency labels assigned to them. Can you please guide me how should I train a Distil-LESA-BERT-6 on my data? I looked at the readme.md which is pointing me to use script_glue_label.sh to do the training. However, when I look at the script, it looks like the model is being trained on MRPC task and not on the urgency classification task. Which script should I run to train LESA-BERT on urgency detection task, and then which script should I run to distill such trained model?

Also the paper mentions that to train LESA BERT, we should first start with pre-trained BioBERT. However, in the script, I see that only BERT-base weights are used as the starting point. Can you please clarify?

Thanks in advance!

**Comment by kdkalp** [comment]
Hi Shijing,

Thank you for a quick response. 
I am looking to tryout LESA-BERT on a unique medical messages data, specifically looking for Distil-LESA-BERT-6 that is referred in your very insightful [paper](http://proceedings.mlr.press/v126/si20a.html) (table 3)  for urgency detection on medical  messages. 

So I understand that I need to initialize with a pre-trained BERT model and then train LESA-BERT on a small amount of text data labeled with urgency categories. Then I need to Distill such trained LESA-BERT to a variant with 6 encoder layers. 

Can you please point me to the scripts to execute these two steps - first the training of LESA-BERT and then Distillation of such trained model to 6 layers? Also, for training LESA-BERT, should I initialize the model weights with BioBERT or with BERT-base?

Thank you so much for your help!

**Comment by kdkalp** [comment]
Just wanted to add -- 
I looked at the script_glue_label.sh that is mentioned in the Readme.md. However that script seems to be training for MRPC task, not for urgency detection. Also I couldn't find the code to distill the LESA-BERT model.
